x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.test.ts

```diff
@@ -230,6 +225,14 @@ describe('Observability AI Assistant client', () => {
               reject();
             };
           });
+        })
```

Comment:
Since I added a new Promise to resolve the instructions, the Promise for the title seems to resolve first, so the first call to the LLM is for the title while the second call is the prompt itself.
Hence we needed to flip these two mock steps to get the right reply to the right place.